plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceRuntimeInfo.java

```diff
@@ -315,11 +318,21 @@ public String projectsRoot() {
             String portProtocol = portEntry.getKey();
             // we are assigning ports automatically, so have 1 to 1 binding (at least per protocol)
             String externalPort = portEntry.getValue().get(0).getHostPort();
+
+            // If we are sending messages to the container directly, we don't want to use the mapped ports.
+            String internalHostnameAndPort;
+            if (useInternalAddress) {
+                String internalPort = portProtocol.split("/")[0];
+                internalHostnameAndPort = internalHostname + ":" + internalPort;
```

Comment:
In that case it is supposed that browser will connect to ephemeral port or exposed port? I'm asking because as far as I remember it should be exposed port, but I'm not sure.